### PR TITLE
CVSL-1852 Adding release in next 2 days flag

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt [hunk 1]
```diff
@@ -42,6 +42,13 @@ class ReleaseDateService(
     return actualReleaseDate < 1.workingDaysBefore(conditionalReleaseDate)
   }
 
+  fun isDueToBeReleasedInTheNextTwoWorkingDays(sentenceDateHolder: SentenceDateHolder): Boolean {
+    val actualReleaseDate = sentenceDateHolder.actualReleaseDate
+    val conditionalReleaseDate = sentenceDateHolder.conditionalReleaseDate
+    val earliestReleaseDate = listOfNotNull(actualReleaseDate, conditionalReleaseDate).minOrNull() ?: return false
+    return LocalDate.now(clock) > 2.workingDaysBefore(earliestReleaseDate) && LocalDate.now(clock) <= earliestReleaseDate
```
Comment (SebNegahbanMoJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt:49]: Should this be `LocalDate.now(clock) >= 2.workingDaysBefore(earliestReleaseDate)`? Or is it like JS where `LocalDate.now() > LocalDate.now().startOfDay()` is true?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt [hunk 2]
```diff
@@ -352,7 +352,90 @@ class ReleaseDateServiceTest {
...
+      )
+
+      assertThat(service.isDueToBeReleasedInTheNextTwoWorkingDays(licence)).isTrue
+    }
+
+    @Test
+    fun `licence is due to be released in the next two working days when CRD is 2 working days in the future`() {
+      val licence = createCrdLicence().copy(
+        actualReleaseDate = null,
+        conditionalReleaseDate = LocalDate.now(clock).plusDays(3),
```
Comment (SebNegahbanMoJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt:405]: Should this not be `.plusDays(2)`?

Also perhaps more importantly, should this not be failing in its current state?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt [hunk 3]
```diff
@@ -352,7 +352,90 @@ class ReleaseDateServiceTest {
...
+      )
+
+      assertThat(service.isDueToBeReleasedInTheNextTwoWorkingDays(licence)).isTrue
+    }
+
+    @Test
+    fun `licence is due to be released in the next two working days when CRD is more than 2 working days in the future`() {
```
Comment (SebNegahbanMoJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt:412]: Minor, not worth its own commit, but
`licence is **not** due to be released ...`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt (hunk 2 above)
Comment (andrewrlee) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt:405]: hate these tests but now == `2023-11-10T00:00:00Z` which is a Saturday 
So 3 days is 2 working days

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt (hunk 3 above)
Comment (andrewrlee) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt]: 👍  fixed this

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt (hunk 1 above)
Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt]: LocalDate doesn't have a time component so `LocalDate.now() > LocalDate.now().startOfDay()` is not true.

but I rewrote the tests and yes it should have been `>=`

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt (hunk 2 above)
Comment (andrewrlee) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt]: I've rewritten these

**Pull request merged**